### ISSUE TITLE
Wrong Test Path

### ISSUE THREAD

**Issue by Engineerumair** [fatemehsrz/Shortest_Distance#1]
I saw the path for test data set in generate_train_test.py is the following :

f2= open('./data/%s_test.txt'%(graph_name), 'w') 

But I found the following in feedforward.py :
f2= open('%s_truth.txt'%(graph_name), 'r')

**Comment by liyue123-jpg** [comment]
I can't find “embeddings_file” in this project：
embeddings_file = "./../emb_deepwalk/%s_%d.deepwalk"%(graph_name, emb_size)
model = word2vec.KeyedVectors.load_word2vec_format(embeddings_file, binary=False)

**Comment by Engineerumair** [comment]
> I can't find “embeddings_file” in this project： embeddings_file = "./../emb_deepwalk/%s_%d.deepwalk"%(graph_name, emb_size) model = word2vec.KeyedVectors.load_word2vec_format(embeddings_file, binary=False)

You should generate embedding file yourself and give that path to the project.

**Comment by fatemehsrz** [comment]
yes, exactly, you need to apply a node to vector model yourself first, then approximate the distance by FFN

**Comment by fatemehsrz** [comment]
you can try the euclidean distance ||v1-v2|| as an additional experiment

**Comment by Engineerumair** [comment]
> yes, exactly, you need to apply a node to vector model yourself first, then approximate the distance by FFN

Fatemeh, I have applied the same work on weighted graphs but the accuracy is very less, I found you considered the shortest nodes of cost between 2 and 6 only as the dataset were having equal distribution of these weights.

In my case the when I assign weighted between 1 and 10 and when tries to generate train and test set with weighted cost the cost exceeds 10 as there are different nodes added to the path with different weights.

I think the embedding trained between 1 and 10 weights where FFN trained on higher cost , may be this is the problem where it can't predict well.

Do you suggest something for weighted graphs to be trained on?

**Comment by fatemehsrz** [comment]
my method is just for unweighted graphs.
For the weighted graphs, you should apply a weighted node to vector model first. Then, approximate the distance using other NN models like Siamese networks. Or even think about building your own NN model that works better!

**Comment by Engineerumair** [comment]
Yeah, I have used weighted graph and generated training pairs for weighted
graph using Dijkstra rather than BFS.

Should I use embedding on graph or training set because some of the
training set cost are not in weighted graph, what do you suggest.

**Comment by fatemehsrz** [comment]
there are also embedding techniques  particularly for weighted graphs, first use them

**Comment by Engineerumair** [comment]
I have used node2vec+ embedding which is used for weighted graphs.

**Comment by Engineerumair** [comment]
Is this essential to learn embeddings first on graph then train neural
network on training set, shouldn't we train both embedding and neural
network on training set?

**Comment by Engineerumair** [comment]
I also get good accuracy when I use training set of cost between 1 and 5.

**Comment by Engineerumair** [comment]
Can I have 30 minutes voice meeting over zoom to share my results with you?
